File: 2021/1/1.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <string>
#include <vector>

int main() {
    std::fstream f("sea_depth.txt");
    std::vector<int> values;
    std::string str;
    long long j = 0, k = 0, last = 0;
    while (std::getline(f, str)) {
        values.push_back(std::stoi(str));
        if (values.size() == 1 && values[values.size() - 2] < values[values.size() - 1])
            ++j;

        if (values.size() <= 3) continue;
        if (values.size() == 3)
            last = values[0] + values[1] + values[2];
        int l = values.size() - 1;
        long long sum = values[l - 2] + values[l - 1] + values[l];
        if (sum > last)
            ++k;
        last = sum;
    }

    std::cout << j << std::endl;
    std::cout << k << std::endl;
}

File: 2021/1/1.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final s = File("sea_depth.txt")
    .openRead()
    .map(utf8.decode)
    .transform(LineSplitter());

  var j = 0, k = 0, last = 0;
  final l = <int>[];
  await for (final ln in s) {
    l.add(int.parse(ln));
    // Part one
    if (l.length > 1 && l[l.length - 2] < l[l.length - 1])
      ++j;

    // Part two
    if (l.length <= 3) continue;
    if (l.length == 3)
      last = l.sublist(0, 3).fold(0, (p, n) => p + n);
    final sum = l.sublist(l.length - 3).fold(0, (p, n) => n + (p as int));
    if (sum > last)
      ++k;
    last = sum;
  }
  
  print(j); print(k);
}
